libs/components/layout/testing/src/fluid-grid/column-harness.ts

```diff
@@ -0,0 +1,98 @@
+import { HarnessPredicate } from '@angular/cdk/testing';
+import { SkyComponentHarness } from '@skyux/core/testing';
+
+import { SkyColumnHarnessFilters } from './column-harness-filters';
+
+const sizes = ['xs', 'sm', 'md', 'lg'];
+
+/**
+ * Harness for interacting with a fluid grid column component in tests.
+ */
+export class SkyColumnHarness extends SkyComponentHarness {
+  /**
+   * @internal
+   */
+  public static hostSelector = 'sky-column';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a
+   * `SkyColumnHarness` that meets certain criteria
+   */
+  public static with(
+    filters: SkyColumnHarnessFilters,
+  ): HarnessPredicate<SkyColumnHarness> {
+    return SkyColumnHarness.getDataSkyIdPredicate(filters);
+  }
+
+  /**
+   * Gets the size of the column in an XSmall responsive context.
+   */
+  public async getXSmallSize(): Promise<number> {
+    return this.#getColumnSize(0);
+  }
+
+  /**
+   * Gets the size of the column in a Small responsive context.
+   */
+  public async getSmallSize(): Promise<number> {
+    return this.#getColumnSize(1);
+  }
+
+  /**
+   * Gets the size of the column in a Medium responsive context.
+   */
+  public async getMediumSize(): Promise<number> {
+    return this.#getColumnSize(2);
+  }
+
+  /**
+   * Gets the size of the column in a Large responsive context.
+   */
+  public async getLargeSize(): Promise<number> {
+    return this.#getColumnSize(3);
+  }
+
+  async #getColumnSize(index: number): Promise<number> {
+    const size = sizes[index];
+
+    const result = await this.#getColumnClass(size);
+
+    /* istanbul ignore if */
+    if (!result && !index) {
```

Comment:
is this not a testable scenario?

Reply:
No. There should always be a class to fall back to. If we get a result of 0 after running the function against `xsmall`, which has an index of 0 and should always have a positive value, then something went wrong and without this exception the function would never stop looping.